sdk/openai/Azure.AI.OpenAI/src/Custom/AsyncAutoResetEvent.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.AI.OpenAI.Custom
+{
+    // Adapted from https://devblogs.microsoft.com/pfxteam/building-async-coordination-primitives-part-2-asyncautoresetevent/
+    internal class AsyncAutoResetEvent
```

Comment:
In general classes in the `Custom` folder are meant to mirror `Generated` classes.  This class can go in root or a sub folder named whatever we like.

Reply:
Got it! I'll move the new, non-mirrored files into a folder named "Streaming." They may move back later when we can generate a lot of it, but better to be correct at every stage.